File: app/src/androidTest/java/com/futsch1/medtimer/DeleteTest.java
package com.futsch1.medtimer;


import static com.adevinta.android.barista.assertion.BaristaListAssertions.assertListItemCount;
import static com.adevinta.android.barista.interaction.BaristaClickInteractions.clickOn;
import static com.adevinta.android.barista.interaction.BaristaListInteractions.clickListItem;
import static com.adevinta.android.barista.interaction.BaristaListInteractions.clickListItemChild;
import static com.adevinta.android.barista.interaction.BaristaMenuClickInteractions.openMenu;

import org.junit.Test;

public class DeleteTest extends BaseTestHelper {

    @Test
    //@AllowFlaky(attempts = 1)
    public void testDelete() {
        openMenu();
        clickOn(R.string.generate_test_data);

        AndroidTestHelper.navigateTo(AndroidTestHelper.MainMenu.MEDICINES);

        clickListItem(R.id.medicineList, 0);

        openMenu();
        clickOn(R.string.delete);
        clickOn(R.string.yes);
        assertListItemCount(R.id.medicineList, 3);

        clickListItem(R.id.medicineList, 2);
        clickListItemChild(R.id.reminderList, 1, R.id.openAdvancedSettings);

        openMenu();
        clickOn(R.string.delete);
        clickOn(R.string.yes);
        assertListItemCount(R.id.reminderList, 1);
    }

}

File: app/src/androidTest/java/com/futsch1/medtimer/DeleteTest.kt
package com.futsch1.medtimer

import com.adevinta.android.barista.rule.flaky.AllowFlaky
import com.futsch1.medtimer.core.ui.R
import dagger.hilt.android.testing.HiltAndroidTest
import org.junit.Test


@HiltAndroidTest
class DeleteTest : MedTimerTestBase() {
    @Test
    @AllowFlaky(attempts = 3)
    fun testDelete() {
        menus.clickAppOption(R.string.generate_test_data)

        navigation.toMedicines()
        medicines.assertCount(4)

        medicines.clickNamed(OMEGA_3)
        menus.clickEditMedicineOption(R.string.delete)
        dialogs.confirm()

        medicines.assertCount(3)
        medicines.assertNameNotContains(OMEGA_3)

        medicines.clickNamed(SELEN)
        reminders.assertCount(2)

        reminders.delete(1)

        reminders.assertCount(1)
    }

    companion object {
        private const val OMEGA_3 = "Omega 3 (EPA/DHA 500mg)"
        private const val SELEN = "Selen (200 µg)"
    }
}
